.github/workflows/tag-and-build-release.yaml

```diff
@@ -0,0 +1,108 @@
+name: Tag and Build Release
```

Comment:
I used to trigger releases by pushing tags, how does this one get triggered?

Reply:
You just enter a version number like 0.0.0
This action tags the ref you're on (probably main), then builds it, and publishes the build artifacts to a new release.

It's one giant manually triggered pipeline.

Reply:
see: https://github.com/loosebazooka/slsa-java-test/actions/runs/2950354692